test/integration/testArrayBindCustomerTable.js

```diff
@@ -1,66 +1,57 @@
-	'APP_ID VARCHAR(16777216), OCCURREDAT VARCHAR(16777216), SHOP_ID VARCHAR(16777216),' +
-	'TYPE VARCHAR(16777216),ID VARCHAR(16777216),CHARGE_AMOUNT_AMOUNT FLOAT,' +
-	'CHARGE_AMOUNT_CURRENCYCODE VARCHAR(16777216),CHARGE_ID VARCHAR(16777216),' +
-	'CHARGE_NAME VARCHAR(16777216),CHARGE_TEST VARCHAR(16777216),CHARGE_BILLINGON VARCHAR(16777216),' +
-	'REASON VARCHAR(16777216),DESCRIPTION VARCHAR(16777216),APPCREDIT_AMOUNT_AMOUNT FLOAT,' +
-	'APPCREDIT_AMOUNT_CURRENCYCODE VARCHAR(16777216),APPCREDIT_ID VARCHAR(16777216),'+
-	'APPCREDIT_NAME VARCHAR(16777216),APPCREDIT_TEST VARCHAR(16777216),APP_NAME VARCHAR(16777216),'+
-	'SHOP_MYSHOPIFYDOMAIN VARCHAR(16777216),SHOP_NAME VARCHAR(16777216),APP_APIKEY VARCHAR(16777216))';
-  var insertWithQmark = 'insert into EVENTS_TEMP values(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)';
+  let connection;
+  const tableName = 'EVENTS_TEMP';
+  const createTable = `create or replace TABLE ${tableName} (ORGANIZATION_ID VARCHAR(16777216),` +
+    'APP_ID VARCHAR(16777216), OCCURREDAT VARCHAR(16777216), SHOP_ID VARCHAR(16777216),' +
+    'TYPE VARCHAR(16777216),ID VARCHAR(16777216),CHARGE_AMOUNT_AMOUNT FLOAT,' +
+    'CHARGE_AMOUNT_CURRENCYCODE VARCHAR(16777216),CHARGE_ID VARCHAR(16777216),' +
+    'CHARGE_NAME VARCHAR(16777216),CHARGE_TEST VARCHAR(16777216),CHARGE_BILLINGON VARCHAR(16777216),' +
+    'REASON VARCHAR(16777216),DESCRIPTION VARCHAR(16777216),APPCREDIT_AMOUNT_AMOUNT FLOAT,' +
+    'APPCREDIT_AMOUNT_CURRENCYCODE VARCHAR(16777216),APPCREDIT_ID VARCHAR(16777216),' +
+    'APPCREDIT_NAME VARCHAR(16777216),APPCREDIT_TEST VARCHAR(16777216),APP_NAME VARCHAR(16777216),' +
+    'SHOP_MYSHOPIFYDOMAIN VARCHAR(16777216),SHOP_NAME VARCHAR(16777216),APP_APIKEY VARCHAR(16777216))';
+  const insertWithQmark = `insert into ${tableName} values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`;
 
-  before(function (done)
+  before(async () =>
   {
     connection = testUtil.createConnection();
-    testUtil.connect(connection, function ()
-    {
-      connection.execute({
-        sqlText: createTable,
-        complete: function (err)
-        {
-          testUtil.checkError(err);
-          done();
-        }
-      });
-    });
+    await testUtil.connectAsync(connection);
+    await testUtil.executeCmdAsync(connection, createTable);
   });
 
-  after(function (done)
+  after(async () =>
   {
-    testUtil.destroyConnection(connection, done);
+    await testUtil.executeCmdAsync(connection, `DROP TABLE IF EXISTS ${tableName};`);
+    await testUtil.destroyConnectionAsync(connection);
   });
 
   it('testArrayBindCustomerTable', function (done)
   {
-    var arrBind = [];
-    var count = 100000;
-    for(var i = 0; i<count; i++)
+    const arrBind = [];
+    const rowsToInsert = 10000; // 100000 rows causes "Statement exhausted compilation resources and was canceled. Please contact support." in regression environment
```

Comment:
Probably you tried investigate what does mean "Statement exhausted compilation resources". It is a problem of test environment?

Reply:
It seems to be problem of the regression environment. There is a question if we need so many records added in this test to check the behaviour. Maybe this test should be part of performance tests in the future.